File: client/Create.js
// Test for CrudReference project, POST command
// usage:
//  node Create "post this data"

var http = require('http')
apiHost = 'localhost',
apiPath = '/api/items',
apiPort = 47514;

function PostData(data) {
    data = '{"Content":"' + data + '"}';
    var options = {
        host: apiHost,
        path: apiPath,
        port: apiPort,
        method: 'POST',
        headers: {
            'Content-Type': 'application/json',
            'Content-Length': data.length
        }
    };
    var request = http.request(options, function (res) {
        var loc = res.headers.location;
        console.log("SUCCESS. Location:", loc);
        result = true;
    });
    request.on('error', function (err) {
        console.log("Error:", err.code);
    });

    request.write(data);
    request.end();
}

if (3 != process.argv.length) {
    console.log("### Usage: node Create data");
    console.log("### Example: node Create 'Test Data'");
} else {
    PostData(process.argv[2]);
}

File: client/python/Create.py
import httplib

def api_create(data):
    data = '{"Content":"' + data + '"}';
    conn = httplib.HTTPConnection("localhost", 47514)
    conn.connect()
    request = conn.putrequest('POST', '/api/item')
    headers = {}
    headers['Content-Type'] = 'application/json'
    headers['Content-Length'] = len(data)
    for k in headers:
        conn.putheader(k, headers[k])
    conn.endheaders()

    conn.send(data)
    resp = conn.getresponse()
    if 201 == resp.status:
        loc = resp.getheader("location")
        print "SUCCESS: Location =", loc
    else:
        print "ERROR:", resp.status, resp.reason
    conn.close()

api_create("Testing 1, 2, 3")
